src/ios/NativeStorage.m

```diff
@@ -211,19 +211,23 @@ - (void) setItem: (CDVInvokedUrlCommand*) command
 		NSString* reference = [command.arguments objectAtIndex:0];
 		NSString* aString = [command.arguments objectAtIndex:1];
 
-		if(reference!=nil && [aString class] == [NSNull class])
+		if(reference==nil)
+		{
+			pluginResult = [CDVPluginResult resultWithStatus: CDVCommandStatus_ERROR messageAsString:@"Reference was null"];
```

Comment:
I think introducing messageAsString here is a breaking change as the original behavior for a nil reference was the messageAsInt.
I think it makes more sense to just change the '==' to '!=' on the null check of the original code to match the intended behavior.
If you flip the logic like you did here then 
```
if(reference==nil || [aString class] == [NSNull class])
{
pluginResult = [CDVPluginResult resultWithStatus: CDVCommandStatus_ERROR messageAsInt:3];
}
else ...
```

Reply:
Fair enough; I'll keep it as an int error code. 

IMO, it would be beneficial to introduce a separate code or message to indicate specifically which parameter was null.